.github/workflows/publish_to_test_pypi.yaml

```diff
@@ -0,0 +1,25 @@
+name: Publish Test PyPI Release
+
+on:
+  push:
+    tags: ["*"]
```

Comment:
Are we sure we don't want to scope this a little more narrowly?

Reply:
I considered it. We could use `*.*.*` or something else like that.
But we don't have any other tags that we create in this repo as part of our normal operating procedures, and since this workflow only publishes to test-pypi, I shrugged it off for now.

I'm happy to revisit this pattern in the future, but I didn't want to block here on figuring out what kinds of regexes are supported.